sdcm/logcollector.py

```diff
@@ -0,0 +1,819 @@
+                                  search_locally=True),
+                    CommandLogEntity(name='cpu_info',
+                                     command='cat /proc/cpuinfo'),
+                    CommandLogEntity(name='mem_info',
+                                     command='cat /proc/meminfo'),
+                    CommandLogEntity(name='interrupts',
+                                     command='cat /proc/interrupts'),
+                    CommandLogEntity(name='vmstat',
+                                     command='cat /proc/vmstat'),
+                    CommandLogEntity(name='scylla.yaml',
+                                     command='cat /etc/scylla/scylla.yaml'),
+                    CommandLogEntity(name='coredumps.info',
+                                     command='sudo coredumpctl info')
+                    ]
+    cluster_log_type = "db-cluster"
+
+
+class LoaderLogCollector(LogCollector):
+    cluster_log_type = "loader-set"
+
+
+class MonitorLogCollector(LogCollector):
+    """SCT monitor set log collector
+
+    Collect required log entites for Monitor set nodes
+
+    Extends:
+        LogCollector
+
+    """
+    log_entities = [
+        CommandLogEntity(name='aprom.log',
+                         command='docker logs --details -t aprom'),
+        CommandLogEntity(name='agraf.log',
+                         command='docker logs --details -t agraf'),
+        CommandLogEntity(name='aalert.log',
+                         command='docker logs --details -t aalert'),
+        FileLogEntity(name='scylla_manager.log',
+                      command='sudo journalctl -u scylla-manager.service --no-tail',
+                      search_locally=True),
+        PrometheusSnapshotsEntity(name='prometheus_data'),
+        MonitoringStackEntity(name='monitoring-stack'),
+        GrafanaScreenShotEntity(name='grafana-screenshot'),
+        GrafanaSnapshotEntity(name='grafana-snapshot')
+    ]
+    cluster_log_type = "monitor-set"
+
+
+class SCTLogCollector(LogCollector):
+    """logs for hydra test run
+
+    Find and collect local version of files
+    for sct
+
+    Extends:
+        LogCollector
+
+    """
+    log_entities = [
+        FileLogEntity(name='sct.log',
+                      search_locally=True),
+        FileLogEntity(name='raw_events.log',
+                      search_locally=True),
+        FileLogEntity(name='events.log',
+                      search_locally=True),
+        FileLogEntity(name='output.log',
+                      search_locally=True),
+        FileLogEntity(name='critical.log',
+                      search_locally=True)
+    ]
+    cluster_log_type = 'sct-runner'
+
+    def collect_logs(self, local_search_path=None):
+        for ent in self.log_entities:
+            ent.collect(None, self.local_dir, None, local_search_path=local_search_path)
+        # TODO: rewrite to zipfile module with zip_64
```

Comment:
yep this is impotent since our files might get big

Reply:
Done